### PR TITLE
Tasks is due

### PR DESCRIPTION
**Pull request by TheHollidayInn**
### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

- Adds isDue Field
- Computes isDue on create, update, score and cron

### CODE REVIEW

File: test/api/v3/integration/tasks/POST-tasks_id_score_direction.test.js [hunk 1]
```diff
@@ -208,6 +208,13 @@ describe('POST /tasks/:id/score/:direction', () => {
       expect(task.completed).to.equal(false);
     });
 
+    it('computes is due', async () => {
```
Comment (khipkin) [inline test/api/v3/integration/tasks/POST-tasks_id_score_direction.test.js]: nit: 'computes isDue'

File: test/api/v3/unit/libs/cron.test.js [hunk 2]
```diff
@@ -358,6 +358,14 @@ describe('cron', () => {
       };
     });
 
+    it('computes isdue', () => {
```
Comment (khipkin) [inline test/api/v3/unit/libs/cron.test.js]: nit: 'computes isDue'

**Review by khipkin — APPROVED** [review]

File: website/server/models/task.js [hunk 3]
```diff
@@ -242,6 +242,7 @@ export let DailySchema = new Schema(_.defaults({
   streak: {type: Number, default: 0},
   daysOfMonth: {type: [Number], default: []}, // Days of the month that the daily should repeat on
   weeksOfMonth: {type: [Number], default: []}, // Weeks of the month that the daily should repeat on
+  isDue: {type: Boolean},
```
Comment (paglias) [inline website/server/models/task.js:245]: should be added to the non updatable lists

**Review by SabreCat — APPROVED** [review]

File: test/api/v3/integration/tasks/POST-tasks_id_score_direction.test.js [hunk 4]
```diff
@@ -208,6 +208,13 @@ describe('POST /tasks/:id/score/:direction', () => {
       expect(task.completed).to.equal(false);
     });
 
+    it('computes isDue', async () => {
+      await user.post(`/tasks/${daily._id}/score/up`);
+      let task = await user.get(`/tasks/${daily._id}`);
+
+      expect(task.isDue).to.equal(true);
```
Comment (crookedneighbor) [inline test/api/v3/integration/tasks/POST-tasks_id_score_direction.test.js:215]: Why is isDue true after it has been scored?

**Review by crookedneighbor — APPROVED** [review]

File: test/api/v3/integration/tasks/POST-tasks_id_score_direction.test.js (hunk 4 above)
Comment (TheHollidayInn) [reply test/api/v3/integration/tasks/POST-tasks_id_score_direction.test.js:215]: isDue has nothing to do with completed as it only represents shouldDo. isDue is combined with a completed check on the UI.

**Pull request merged**